packages/react-components/react-rating-preview/src/components/RatingItem/useRatingItem.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { getPartitionedNativeProps, slot } from '@fluentui/react-utilities';
+import { useFocusWithin } from '@fluentui/react-tabster';
+import type { RatingItemProps, RatingItemState } from './RatingItem.types';
+import { useRatingContextValue_unstable } from '../../contexts/RatingContext';
+import { StarRegular, StarFilled, StarHalfRegular } from '@fluentui/react-icons';
+
+/**
+ * Create the state required to render RatingItem.
+ *
+ * The returned state can be modified with hooks such as useRatingItemStyles_unstable,
+ * before being passed to renderRatingItem_unstable.
+ *
+ * @param props - props from this instance of RatingItem
+ * @param ref - reference to root HTMLElement of RatingItem
+ */
+export const useRatingItem_unstable = (props: RatingItemProps, ref: React.Ref<HTMLInputElement>): RatingItemState => {
+  const context = useRatingContextValue_unstable();
+  const { value = 0 } = props;
+
+  const ratingValue = context.value || 0;
+
+  const displayedRatingValue = context.hoveredValue !== undefined ? context.hoveredValue : ratingValue;
```

Comment:
nit, a little less code

```suggestion
  const displayedRatingValue = context.hoveredValue ?? ratingValue;
```